src/Common/src/TypeSystem/Common/GenericParameterDesc.cs

```diff
@@ -20,5 +22,35 @@ public abstract partial class GenericParameterDesc : TypeDesc
         /// Gets the zero based index of the generic parameter within the declaring type or method.
         /// </summary>
         public abstract int Index { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this parameter is covariant.
+        /// </summary>
+        public abstract bool IsCovariant { get; }
```

Comment:
This may be better as a single bit mask. If you need constrains, you pretty much always need all of them.